superset/views/core.py

```diff
@@ -1008,7 +1008,7 @@ def shortner(self):
         db.session.add(obj)
         db.session.commit()
         return Response(
-            "{scheme}://{request.headers[Host]}/r/{obj.id}".format(
+            "{request.headers[origin]}/r/{obj.id}".format(
```

Comment:
I am ok to use `ORIGIN` instead of `HOST`, why remove `{scheme}` part?

Also please check the error message in unit test, you probably need to find a way to get HTTP_ORIGIN header

Reply:
i will double check on the schema but in my local testing it was getting added automatically.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin